xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -426,6 +496,111 @@ private String processRouteConfig(RouteConfiguration config) {
     return null;
   }
 
+  /**
+   * Handles EDS response, which contains a list of ClusterLoadAssignment messages. Each
+   * ClusterLoadAssignment message contains information for endpoints in a cluster, grouped
+   * by localities. The response is NACKed if contains invalid information for gRPC's usage.
+   * Otherwise, an ACK request is sent to management server. Information in ClusterLoadAssignment
+   * messages are saved in local cache, in case of corresponding endpoint watchers are registered
+   * later. Endpoint watchers interested in endpoint information contained in this response are
+   * notified.
+   */
+  private void handleEdsResponse(DiscoveryResponse edsResponse) {
+    logger.log(Level.FINE, "Received an EDS response: {0}", edsResponse);
```

Comment:
`toString()` will generate useless info - why not write a helper function to get useful info out of the response as a single string?

Reply:
We've discussed the issue for logging responses offline. We will handle this in a separate PR to fix everything about logging with JSON formatter in the future.